TestFoundation/TestNSURLSession.swift

```diff
@@ -608,3 +626,61 @@ class CustomProtocol : URLProtocol {
         return
     }
 }
+
+class HTTPRedirectionDataTask : NSObject {
```

Comment:
Can we inherit from DataTask here? That will possibly avoid some code duplication.